src/core/macros.systemd.in

```diff
@@ -112,7 +112,7 @@ SYSTEMD_INLINE_EOF \
 #   %files
 #   %{_sysusersdir}/%{name}.conf
 %sysusers_create_package() \
-systemd-sysusers --replace=%_sysusersdir/%1.conf - <<SYSTEMD_INLINE_EOF >/dev/null 2>&1 || : \
+[ -x @bindir@/systemd-sysusers] && @bindir@/systemd-sysusers --replace=%_sysusersdir/%1.conf - <<SYSTEMD_INLINE_EOF || : \
```

Comment:
Unfortunately sysusers and tmpfiles can warn about issues that stem from errors in other packages. So if this change went in, it's likely that users will get warnings, and those warnings will appear to come from package A but be really from package B.

Reply:
An interesting point. It begs the question how these macros are supposed to be invoked.

It was a design decision that commands like sysusers and tmpfiles support processing not the entire system configuration, but only the files specified as arguments.
The conclusions I would draw from that are:

1. that every rpm scriptlet is ideally supposed to always call sysusers with _at least one_ argument, and thereby only revealing errors pertaining to files newly installed by the package, and
2. that the invocation of %sysusers_xxx with 0 arguments is considered a logical error, but which is not grave enough to warrant additional code inside the macro definition to reject the case at build time. The worst that could happen if a user passes 0 arguments is, as you say, that syntax errors from other packages will show up. Which, in itself, might be considered a good thing -- syntax errors should be fixed.

What do you think?

Reply:
The macros were initially designed to be called just for one specific file. But this had the problem that the effect of invocation at package installation time would be markedly different than the normal invocation during system boot. So we changed the macros to behave during package installation as during boot with the configuration provided by the package installed. This of course means that the whole configuration is read.

We *could* add a mode where we warn about errors in just a subset of files, and then use this in macro invocations, but this would require more changes to the code first.

Reply:
Oh, and the macro must always be invoked with exactly one argument. Adding checks that it is invoked properly at package build time would be very good.

Reply:
systemctl supports more than one argument, and that is even documented, so I have adjusted the macro logic to permit such likewise.